src/components/DepartmentsList.jsx

```diff
@@ -2,43 +2,158 @@ import React, { Component } from 'react';
 import { apiRequest } from '../api/apiService';
 import withError from '../HOC/withError';
 import withLoader from '../HOC/withLoader';
-import AddNewItemBtn from './AddNewItemBtn';
 import Department from './Department';
-
+import AddDepartmentForm from '../modalForms/AddDepartmentForm';
+import { AppContext } from '../contexts/AppContext';
+import SearchForm from './SearchForm';
+import SortForm from './SortForm';
 
 class DepartmentsList extends Component {
   state = {
     departments: [],
+    limit: 10,
+    currentPage: 1,
+    lastPage: 1,
+    total: 0,
   };
 
-  async componentDidMount() {
+  getDepartments = async () => {
     this.props.toggleLoader();
-
-    const [departmentsError, departments] = await apiRequest.getDepartments();
+    const { limit, currentPage } = this.state;
+    const [departmentsError, departments] = await apiRequest.getDepartments(
+      limit,
+      currentPage
+    );
     if (!departmentsError) {
-      this.setState({ departments: departments.departments });
+      const { limit, currentPage, total } = departments.departments.results;
+      this.setState({
+        departments: departments.departments.departments,
+        limit,
+        currentPage,
+        total,
+        lastPage: Math.ceil(total / limit),
+      });
     } else {
       this.props.setError(departmentsError);
     }
     this.props.toggleLoader();
+  };
+
+  componentDidMount() {
+    this.getDepartments();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.currentPage !== prevState.currentPage) {
+      this.getDepartments();
+      window.scroll(0, 0)
+    }
```

Comment:
wouldn't consider to put such logic into componentDidUpdate

usually it is used for the cases when some specific prop or inexplicit state part is changed. 
But such kind can simply be handled onPageChange